src/main/java/com/autotune/utils/GenericRestApiClient.java

```diff
@@ -70,39 +73,92 @@ public GenericRestApiClient(DataSourceInfo dataSourceInfo) {
     }
 
     /**
-     * This method appends aueryString with baseURL and returns response in JSON using specified authentication.
+     * This method appends queryString with baseURL and returns response in JSON using specified authentication.
      * @param methodType    Http methods like GET,POST,PATCH etc
      * @param queryString
      * @return Json object which contains API response.
      * @throws IOException
      */
     public JSONObject fetchMetricsJson(String methodType, String queryString) throws IOException, NoSuchAlgorithmException, KeyStoreException, KeyManagementException, FetchMetricsError {
-        System.setProperty("https.protocols", "TLSv1.2");
-        String jsonOutputInString = "";
-        SSLContext sslContext = SSLContexts.custom().loadTrustMaterial((chain, authType) -> true).build();  // Trust all certificates
-        SSLConnectionSocketFactory sslConnectionSocketFactory =
-                new SSLConnectionSocketFactory(sslContext, new String[]{"TLSv1.2"}, null, NoopHostnameVerifier.INSTANCE);
-        try (CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslConnectionSocketFactory).build()) {
+        String jsonResponse;
+        try (CloseableHttpClient httpclient = setupHttpClient()) {
+
             HttpRequestBase httpRequestBase;
             if (methodType.equalsIgnoreCase("GET")) {
                 httpRequestBase = new HttpGet(baseURL + URLEncoder.encode(queryString, StandardCharsets.UTF_8));
             } else {
                 throw new UnsupportedOperationException("Unsupported method type: " + methodType);
             }
+
             // Apply authentication
-            if (authenticationStrategy != null) {
-                String authHeader = authenticationStrategy.applyAuthentication();
-                httpRequestBase.setHeader(KruizeConstants.AuthenticationConstants.AUTHORIZATION, authHeader);
+            applyAuthentication(httpRequestBase);
+
+            LOGGER.info("Executing Prometheus metrics request: {}", httpRequestBase.getRequestLine());
```

Comment:
Please evaluate all LOGGER statements to see if they should really be `info` or `debug`